### PR TITLE
BXMSDOC-3914-master: Added a new chapter about embedded process engine and decision engine using WLS.

### PR DESCRIPTION
**Pull request by kaldesai**
Original Epic : https://issues.jboss.org/browse/BXMSDOC-3702

JIRA: https://issues.jboss.org/browse/BXMSDOC-3914

Rendered Output:
[WLS-RHPAM-7.3](http://file.pnq.redhat.com/~kaldesai/BXMSDOC-3915-WLS-pam/#wls-configure-embedded-engine-proc)
[WLS-RHDM-7.3](http://file.pnq.redhat.com/~kaldesai/BXMSDOC-3914-wls-dm/#wls-configure-embedded-engine-proc)

### CODE REVIEW

File: doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc [hunk 1]
```diff
@@ -0,0 +1,260 @@
+[id='wls-configure-embedded-engine-proc']
+ifdef::PAM[]
+= Configuring embedded {PROCESS_ENGINE} and {DECISION_ENGINE} in {WEBLOGIC}.
+endif::[]
+
+ifdef::DM[]
+= Configuring embedded {DECISION_ENGINE} in {WEBLOGIC}
+endif::[]
+
```
Comment (emmurphy1) [inline doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc]: You need to add context. What is an embedded engine. So something like... An embedded engine is... It enables you to do .... If you use an embedded engine in your {PRODUCT} application, you must configure it. ....

File: doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc [hunk 2]
```diff
@@ -0,0 +1,260 @@
+[id='wls-configure-embedded-engine-proc']
+ifdef::PAM[]
+= Configuring embedded {PROCESS_ENGINE} and {DECISION_ENGINE} in {WEBLOGIC}.
+endif::[]
+
+ifdef::DM[]
+= Configuring embedded {DECISION_ENGINE} in {WEBLOGIC}
```
Comment (emmurphy1) [inline doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc]: an embedded

File: doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+
+<!-- Logging dependency. You can use any logging framework compatible with slf4j. -->
+<dependency>
+  <groupId>ch.qos.logback</groupId>
+  <artifactId>logback-classic</artifactId>
+  <version>${logback.version}</version>
+</dependency>
+----
+
+* For {WEBLOGIC}, user needs to declare the following hibernate dependencies in the `pom.xml` file:
```
Comment (emmurphy1) [inline doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc]: needs to  - do not use, use must
so
, you must declare

Comment (emmurphy1) [reply doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc]: I would reword and remove the note at the bottom:
If you intend to use {PROCESS_ENGINE} in a {WEBLOGIC} {PRODUCT} application,  declare the following hibernate dependencies in the `pom.xml` file:
Note, that what I have written might not be correct so check with your SME to fine tune, but the point is put the condition in the intro statement rather than a note below.

File: doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc (hunk 2 above)
Comment (kaldesai) [reply doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc]: ACK

File: doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc (hunk 1 above)
Comment (kaldesai) [reply doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc]: I will try to add a context about the embedded engine.

File: doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+endif::[]
+
+ifdef::DM[]
+= Configuring an embedded {DECISION_ENGINE} in {WEBLOGIC}
+endif::[]
+
+An embedded engine is a light-weight work-flow engines and rule engine which executes your decisions and business processes. They can be embedded as part of your application or deployed as a service possibly in the cloud.
+
+ifdef::PAM[]
+This section describes how to setup an embedded {PROCESS_ENGINE} and {DECISION_ENGINE} in {WEBLOGIC}.
```
Comment (emmurphy1) [inline doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc]: This procedure describes how to configure...
Would you set up both at the same time? Otherwise, use and instead of or.

File: doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+    <version>${version.org.hibernate-4ee7}</version>
+    </dependency>
+  </dependencies>
+</dependencyManagement>
+----
+endif::[]
++
+[NOTE]
+====
+The current version of the property `version.org.hibernate-4ee7` should be retrieved from the {PRODUCT_BA} BOM Parent.
```
Comment (emmurphy1) [inline doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc]: Copy the current version of the  `version.org.hibernate-4ee7` property from the {PRODUCT_BA} BOM Parent.
What is the {PRODUCT_BA} BOM Parent?

File: doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+    <artifactId>hibernate-core</artifactId>
+    <version>${version.org.hibernate-4ee7}</version>
+    </dependency>
+  </dependencies>
+</dependencyManagement>
+----
+endif::[]
++
+ifdef::DM[]
+. If the {DECISION_ENGINE} is used along with persistence support in your project, you must declare the following hibernate dependencies in the `dependencyManagement` secion of your `pom.xml` file.
```
Comment (emmurphy1) [inline doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc]: As above

File: doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc [hunk 7]
```diff
@@ -0,0 +1,286 @@
+[id='wls-configure-embedded-engine-proc']
+ifdef::PAM[]
+= Configuring an embedded {PROCESS_ENGINE} and {DECISION_ENGINE} in {WEBLOGIC}.
+endif::[]
+
+ifdef::DM[]
+= Configuring an embedded {DECISION_ENGINE} in {WEBLOGIC}
+endif::[]
+
+An embedded engine is a light-weight work-flow engines and rule engine which executes your decisions and business processes. They can be embedded as part of your application or deployed as a service possibly in the cloud.
```
Comment (emmurphy1) [inline doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc]: An embedded engine is a light-weight work-flow and rule engine...
An embedded engine can be part of your {PRODUCT} appliction or they can be deployed as a service, for example in the cloud. Do you mean on OpenShift, or in the cloud generally?

File: doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc [hunk 8]
```diff
@@ -0,0 +1,286 @@
...
+   <version>${kie.version}</version>
+   <extensions>true</extensions>
+  </plugin>
+ </plugins>
+</build>
+----
+
+
+ifdef::PAM[]
+. If the {PROCESS_ENGINE} or the {DECISION_ENGINE} along with persistence support is used in your project, you must declare the following hibernate dependencies in the `dependencyManagement` secion of your `pom.xml` file.
```
Comment (emmurphy1) [inline doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc]: If you use the ....with persistence support in your project, ...

File: doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc (hunk 7 above)
Comment (kaldesai) [reply doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc]: In the cloud, not OpenShift.

File: doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc (hunk 8 above)
Comment (kaldesai) [reply doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc]: Ack

File: doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc [hunk 9]
```diff
@@ -0,0 +1,284 @@
+[id='was-configure-embedded-engine-proc']
```
Comment (MarianMacik) [inline doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc]: Id should be wls-configure...

File: doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc [hunk 10]
```diff
@@ -0,0 +1,276 @@
...
+ifdef::PAM[]
+= Configuring an embedded {PROCESS_ENGINE} and {DECISION_ENGINE} in {WEBLOGIC}
+endif::[]
+
+ifdef::DM[]
+= Configuring an embedded {DECISION_ENGINE} in {WEBLOGIC}
+endif::[]
+
+ifdef::PAM[]
+An embedded engine is a light-weight work flow and rule engine that enables you to execute your decisions and business processes. An embedded engine can be part of a {PRODUCT_PAM} application or it can be deployed as a service, for example in the cloud like OpenShift, Kubernetes, and Docker. You can embed an engine in a {PRODUCT_PAM} application through the API or as a set of contexts and dependency injection (CDI) services.
```
Comment (emmurphy1) [inline doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc]: deployed as a service through OpenShift....

File: doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc [hunk 11]
```diff
@@ -0,0 +1,276 @@
...
+endif::[]
+ifdef::DM[]
+A {DECISION_ENGINE} is a light-weight rule engine that enables you to execute your decisions and business processes. A decision engine can be part of a {PRODUCT_DM} application or it can be deployed as a service, for example in the cloud like OpenShift, Kubernetes, and Docker. You can embed {DECISION_ENGINE} in a {PRODUCT_DM} application through the API or as a set of contexts and dependency injection (CDI) services.
+endif::[]
+
+ifdef::PAM[]
+If you intend to use an embedded engines with your {PRODUCT_PAM} application, you must add Maven dependencies to your project by adding the {PRODUCT_BA} bill of materials (BOM) files to the project’s `pom.xml` file. The {PRODUCT_BA} BOM applies to both {PRODUCT_DM} and {PRODUCT_PAM}. For more information about the {PRODUCT_BA} BOM, see https://access.redhat.com/solutions/3405361[What is the mapping between Red Hat Process Automation Manager and the Maven library version?].
```
Comment (emmurphy1) [inline doc-content/enterprise-only/weblogic/wls-configure-embedded-engine-proc.adoc]: an embedded engine
